lib/miq_expression.rb

```diff
@@ -1606,8 +1616,10 @@ def self.ruby_for_date_compare(col_ruby, col_type, tz, op1, val1, op2 = nil, val
 
   def to_arel(exp, tz)
     operator = exp.keys.first
-
-    field = Field.parse(exp[operator]["field"]) if exp[operator].kind_of?(Hash) && exp[operator]["field"]
+    if exp[operator].kind_of?(Hash)
+      field = Field.parse(exp[operator]["field"]) if exp[operator]["field"]
+      exp[operator]["value"] = Field.parse(exp[operator]["value"]).arel_attribute if exp[operator]["value"] && Field.valid_field?(exp[operator]["value"])
```

Comment:
Could there be any unfortunate edge cases with this approach? i.e. could a value look like a field but actually be a value? I'm thinking that the API for this could benefit from being more explicit anyway. Perhaps something like:

```ruby
{"=" => {"field" => "Vm-foo", "value" => {"field" => "Vm-bar"}}}
```

@gtanzillo any thoughts?